api/python/quilt3/telemetry.py

```diff
@@ -0,0 +1,146 @@
+    def has_connectivity():
+        # TODO: Implement this check. Punting for the near term
+        return True
+
+    @staticmethod
+    def check_telemetry_disabled_in_config():
+        """
+        Check if 'telemetry_disabled' field exists in quilt3 config. If it does, return it. If it does not exist, set
+        it to default value of 'false' (to handle case of current users who predate this config field).
+        """
+        config_value = load_config().get("telemetry_disabled")
+        if config_value is not None:
+            return config_value
+        else:
+            set_config_value("telemetry_disabled", False)
+            return False
+
+
+    @staticmethod
+    def check_telemetry_disabled_by_envvar():
+        # TODO: Potential bug where 'QUILT_DISABLE_USAGE_METRICS=False' doesn't work as expected
+        return bool(os.environ.get(DISABLE_USAGE_METRICS_ENVVAR, False))
+
+
+    @staticmethod
+    def telemetry_is_disabled():
+        disabled_via_config = ApiTelemetry.check_telemetry_disabled_in_config()
+        disabled_via_envvar = ApiTelemetry.check_telemetry_disabled_by_envvar()
+        disabled_due_to_no_connection = not ApiTelemetry.has_connectivity()
+
+        if disabled_via_config:
+            return True
+
+        if disabled_via_envvar:
+            return True
+
+        if disabled_due_to_no_connection:
+            return True
+
+        return False
+
+
+    @staticmethod
+    def cleanup_completed_requests():
+        if ApiTelemetry.telemetry_disabled:
+            return
+
+        # Take this opportunity to clean up any completed requests so that list never gets too large.
+        # Might be better to use a done callback
+        # - https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.add_done_callback
+        with ApiTelemetry.pending_reqs_lock:
+            ApiTelemetry.pending_reqs = [r for r in ApiTelemetry.pending_reqs if not r.done()]
+
+
+    @staticmethod
+    def report_api_use(api_name, python_session_id):
+        if ApiTelemetry.telemetry_disabled:
+            return
+
+        data = {
+            "api_name": api_name,
+            "python_session_id": python_session_id,
+            "telemetry_schema_version": TELEMETRY_SCHEMA_VERSION,
+            'client_type': TELEMETRY_CLIENT_TYPE,
+            'client_version': quilt3_version,
+            'platform': sys.platform,
+            'python_implementation': platform.python_implementation(),
+            'python_version_major': platform.python_version_tuple()[0],
+            'python_version_minor': platform.python_version_tuple()[1],
+            'python_version_patch': platform.python_version_tuple()[2]
+        }
+        print(f"Sending data: {data}")
+        r = ApiTelemetry.session.post(TELEMETRY_URL, json=data, headers={'User-Agent': TELEMETRY_USER_AGENT})
+
+        with ApiTelemetry.pending_reqs_lock:
+            ApiTelemetry.pending_reqs.append(r)
+
+
+    def __call__(self, func):
```

Comment:
You could actually get the function name from `func` instead of passing a string into the constructor

Reply:
The function name isn't quite good enough for analytics